alf/examples/ac_cart_pole.gin

```diff
@@ -10,20 +11,34 @@ ActorCriticAlgorithm.reward_shaping_fn = @reward_scaling
 common.reward_scaling.scale = 0.01
 
 # algorithm config
+observation_spec=@get_observation_spec()
+action_spec=@get_action_spec()
+
+actor/ActorDistributionNetwork.input_tensor_spec=%observation_spec
```

Comment:
Seems we are writing the same input_tensor and output_tensor spec in every gin file... Can we fix input_tensor to observation_spec and output_tensor to action_sepc in python?

Reply:
we remove create_algorithm* and configure algorithm directly, this  increase flexibility and as consquence  it may lost some reusability

Reply:
I know. I mean can we still give a default tensor spec configuration for algorithm class?

Reply:
It's possible to do that for some of alf classes.  But not possible for tf_agents classes.

Reply:
Or we can have a base.gin file that has all the default setup, and later gin files can override them only if needed?

Reply:
One consideration is that we copy gin files to output dir (tensorboard events and checkpoints) but missing included gin files,  so may be keep all configuration in just one gin file is better choice current now